src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -677,5 +1100,197 @@ public String toString() {
+             */
+            public ObjectInputFilter.Status checkInput(FilterInfo info) {
+                return ALLOWED.equals(Objects.requireNonNull(filter.checkInput(info), "status"))
+                        ? ALLOWED
+                        : REJECTED;
+            }
+
+            public String toString() {
+                return "rejectUndecided(" + filter + ")";
+            }
+        }
+
+        /**
+         * Builtin Deserialization filter factory.
+         * The builtin deserialization filter factory provides the
+         * {@link #getSerialFilter static serial filter} when invoked from
+         * {@link ObjectInputStream#ObjectInputStream(InputStream) ObjectInputStream constructors}.
+         * When invoked from {@link ObjectInputStream#setObjectInputFilter(ObjectInputFilter)
+         * to set the stream-specific filter} the requested filter replaces the static serial filter,
+         * unless it has already been set. The stream-specific filter can only be set once,
+         * if it has already been set, {@link IllegalStateException} is thrown.
+         * The builtin deserialization filter factory implements the behavior of earlier versions of
+         * setting the static serial filter in the {@link ObjectInputStream} constructor and
+         * {@link ObjectInputStream#setObjectInputFilter}.
+         *
+         */
+        private static final class BuiltinFilterFactory
+                implements BinaryOperator<ObjectInputFilter> {
+            /**
+             * Returns the ObjectInputFilter to be used for an ObjectInputStream.
+             * This method implements the builtin deserialization filter factory.
+             * If the {@code oldFilter} and {@code newFilter} are null,
+             *     the {@link Config#getSerialFilter()} is returned.
+             * If the {@code oldFilter} is {@code null} and {@code newFilter} is {@code not null},
+             *     the {@code newFilter} is returned.
+             * If the {@code oldFilter} is equal to {@link Config#getSerialFilter},
+             *     the {@code newFilter} is returned.
+             * Otherwise {@code IllegalStateException} exception is thrown.
+             *
+             * <p>This is backward compatible behavior with earlier versions of
+             * {@link ObjectInputStream#setObjectInputFilter},
+             * and the initial filter in the {@link ObjectInputStream} constructor.
+             *
+             * @param oldFilter the current filter, may be null
+             * @param newFilter a new filter, may be null
+             * @return an ObjectInputFilter, the new Filter
+             * @throws IllegalStateException if the {@linkplain ObjectInputStream#getObjectInputFilter() current filter}
+             *       is not {@code null} and is not the JVM-wide filter
+             * @throws SecurityException if there is security manager and the
+             *       {@code SerializablePermission("serialFilter")} is not granted
```

Comment:
Where is this thrown? I don't see it in the implementation of `apply` below.

Reply:
The throws clause is not needed, it is thrown in OIS.setObjectInputFilter.